src/pkg/exporter/js_worker.go

```diff
@@ -0,0 +1,90 @@
+package exporter
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gocraft/work"
+	redislib "github.com/goharbor/harbor/src/lib/redis"
+	"github.com/gomodule/redigo/redis"
+)
+
+const (
+	dialConnectionTimeout = 30 * time.Second
+	dialReadTimeout       = 10 * time.Second
+	dialWriteTimeout      = 10 * time.Second
+	pageItemNum           = 20.0
+)
+
+var (
+	redisPool   *redis.Pool
+	jsClient    *work.Client
+	jsNamespace string
+)
+
+// RedisPoolConfig ...
+type RedisPoolConfig struct {
+	URL               string
+	Namespace         string
+	IdleTimeoutSecond int
+}
+
+// InitRedisCli ...
+func InitRedisCli(rdsCfg *RedisPoolConfig) {
+	pool, err := redislib.GetRedisPool("JobService", rdsCfg.URL, &redislib.PoolParam{
+		PoolMaxIdle:           6,
+		PoolIdleTimeout:       time.Duration(rdsCfg.IdleTimeoutSecond) * time.Second,
+		DialConnectionTimeout: dialConnectionTimeout,
+		DialReadTimeout:       dialReadTimeout,
+		DialWriteTimeout:      dialWriteTimeout,
+	})
+	if err != nil {
+		panic(err)
+	}
+	redisPool = pool
+}
+
+// InitBackendWorker initiate backend worker
+func InitBackendWorker(redisPoolConfig *RedisPoolConfig) {
+	pool, err := redislib.GetRedisPool("JobService", redisPoolConfig.URL, &redislib.PoolParam{
```

Comment:
Why create two Redis pools?

Reply:
With line 34

Reply:
one for redis connection. The other one is reuse of `backednWorker`, so it didn't to implements the same logic on exporter.

Reply:
As it's a ` pool`, I think one is enough if it is used in the same process.

Reply:
As an example, the job service only has one Redis pool.